app/models/minion.rb

```diff
@@ -4,18 +4,43 @@
 
 # Minion represents the minions that have been registered in this application.
 class Minion < ApplicationRecord
+  class NotEnoughMinions < StandardError; end
+  class CouldNotAssignRole < StandardError; end
```

Comment:
An Exception should only be raised in exceptional cases and as such, we shouldn't know when that happens (otherwise we should handle that case too). For example, our code should never call assign_roles with more roles that available minions. If that happens then something is wrong in the code. Also we should always be able to assign a role on salt or we should handle the edge cases. That's why I prefer to let it fail for now. See my [comment](https://github.com/SUSE/pharos/pull/15#discussion_r99125280) on @ereslibre's review too.